src/components/CanvasWrapper.tsx

```diff
@@ -199,15 +294,188 @@ export default function CanvasWrapper(inputProps: CanvasWrapperProps): ReactElem
         props.onTrackClicked(newTrack);
       }
     },
-    [props.dataset]
+    [canv, props.dataset]
+  );
+
+  const vec2ToArray = (v: Vector2): [number, number] => [v.x, v.y];
+
+  /**
+   * Returns the full size of the frame in screen pixels, including offscreen pixels.
+   */
+  const getFrameSizeInScreenPx = useCallback((): [number, number] => {
+    const canvasSizePx = getCanvasSizePx();
+    const frameResolution = props.dataset ? vec2ToArray(props.dataset.frameResolution) : canvasSizePx;
+    const canvasZoom = 1 / canvasZoomInverse.current;
+    return mathUtils.getFrameSizeInScreenPx(canvasSizePx, frameResolution, canvasZoom);
```

Comment:
This callback is a wrapper around the actual `math_utils.getFrameSizeInScreenPx` function. I decided it would be cleaner to pass in all the reused props rather than have to repeat this every time it gets called.